torch/csrc/dynamo/eval_frame.c

```diff
@@ -1161,29 +1208,106 @@ PyObject* torch_c_dynamo_eval_frame_init(void) {
   Py_INCREF(Py_None);
   eval_frame_callback_set(Py_None);
 
-  PyObject* module = PyModule_Create(&_module);
-  if (module == NULL) {
-    return NULL;
-  }
-
-#if IS_PYTHON_3_11_PLUS
+  #if IS_PYTHON_3_11_PLUS
   if (PyType_Ready(&THPPyInterpreterFrameType) < 0) {
-    return NULL;
+    goto failure;
```

Comment:
I don't love gotos, can it be a func?

Reply:
The goto is a CPython habit I picked up. I can change it to just decref before failing :). I'll take note that PyTorch convention doesn't use `gotos` for future contributions!

Reply:
I realized we don't need to decref `Py_None` since it's a singleton and from 3.12 onwards it's immortal anyways (ie fixed refcount so decref/incref is a no-op on it)